### PR TITLE
Fix platform detection for pocl

### PR DESCRIPTION
**Pull request by matthiasdiener**
In pocl, "dev.platform.vendor" is "The pocl project".

### CODE REVIEW

File: pyopencl/characterize/__init__.py [hunk 1]
```diff
@@ -281,12 +281,14 @@ def get_simd_group_size(dev, type_size):
         pass
 
     lc_vendor = dev.platform.vendor.lower()
-    if "nvidia" in lc_vendor:
+    lc_real_vendor = dev.vendor.lower()
```
Comment (inducer) [inline pyopencl/characterize/__init__.py]:
```suggestion
    lc_dev_vendor = dev.vendor.lower()
```

File: pyopencl/characterize/__init__.py [hunk 2]
```diff
@@ -281,12 +281,14 @@ def get_simd_group_size(dev, type_size):
         pass
 
     lc_vendor = dev.platform.vendor.lower()
```
Comment (inducer) [inline pyopencl/characterize/__init__.py]:
```suggestion
    lc_plat_vendor = dev.platform.vendor.lower()
```

**Review by inducer — COMMENTED** [review]
Thanks, good catch. I do object to the variable names though. :)

**Comment by matthiasdiener** [comment]
> Thanks, good catch. I do object to the variable names though. :)

Fixed, thanks.

**Comment by inducer** [comment]
Thanks!

**Pull request merged**